notifier/eventsHandler.go

```diff
@@ -0,0 +1,145 @@
+package notifier
+
+import (
+	"context"
+	"time"
+
+	"github.com/ElrondNetwork/elrond-go-logger/check"
+	"github.com/ElrondNetwork/notifier-go/config"
+	"github.com/ElrondNetwork/notifier-go/data"
+	"github.com/ElrondNetwork/notifier-go/redis"
+)
+
+const (
+	setnxRetryMs       = 500
+	revertKeyPrefix    = "revert_"
+	finalizedKeyPrefix = "finalized_"
+)
+
+// ArgsEventsHandler defines the arguments needed for an events handler
+type ArgsEventsHandler struct {
+	Config    config.ConnectorApiConfig
+	Locker    redis.LockService
+	Publisher Publisher
+}
+
+type eventsHandler struct {
+	config    config.ConnectorApiConfig
+	locker    redis.LockService
+	publisher Publisher
+}
+
+// NewEventsHandler creates a new events handler component
+func NewEventsHandler(args ArgsEventsHandler) (*eventsHandler, error) {
+	err := checkArgs(args)
+	if err != nil {
+		return nil, err
+	}
+
+	return &eventsHandler{
+		locker:    args.Locker,
+		publisher: args.Publisher,
+	}, nil
+}
+
+func checkArgs(args ArgsEventsHandler) error {
+	if check.IfNil(args.Locker) {
+		return ErrNilLockService
+	}
+	if check.IfNil(args.Publisher) {
+		return ErrNilPublisherService
+	}
+
+	return nil
+}
+
+// HandlePushEvents will handle push events received from observer
+func (eh *eventsHandler) HandlePushEvents(events data.BlockEvents) {
+	shouldProcessEvents := true
+	if eh.config.CheckDuplicates {
+		shouldProcessEvents = eh.tryCheckProcessedOrRetry(events.Hash)
+	}
+
+	if events.Events != nil && shouldProcessEvents {
+		log.Info("received events for block",
+			"block hash", events.Hash,
+			"shouldProcess", shouldProcessEvents,
+		)
+		eh.publisher.BroadcastChan() <- events
+	} else {
+		log.Info("received duplicated events for block",
```

Comment:
all these prints will return either 
`ignoring = true` or
`shouldProcess = true` which are read badly. 
Please change them in a consistent way, something like `will process = true` and `processed = false`

Reply:
changed log prints